v7.4.0 (#13058)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.4.0

> _May 9, 2024_

Reviewer: ```suggestion
_May 10, 2024_
```

We'd like to offer a big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:

- ✨ Add optional `id` attribute on shortcut items of the Date and Time Pickers

> - 🎁 Add support for Jalaali v3 in the Date and Time Pickers

Reviewer 2: ```suggestion
- 🎁 Add support for `date-fns-jalali` v3 in the Date and Time Pickers
```

Otherwise it can be confusing with Moment Jalaali

- 🐞 Bugfixes
- 📚 Documentation improvements

### Data Grid

#### `@mui/x-data-grid@7.4.0`

- [DataGrid] Fix error when focus moves from column header to `svg` element (#13028) @oukunan
- [DataGrid] Fix error on column groups change (#12965) @romgrk

#### `@mui/x-data-grid-pro@7.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.4.0`.

#### `@mui/x-data-grid-premium@7.4.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.4.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@7.4.0`

- [fields] Fix regression preventing form submit on "Enter" click (#13065) @LukasTy
- [pickers] Add `AdapterDateFnsJalaliV3` adapter (#12891) @smmoosavi
- [pickers] Add optional `id` attribute on shortcut items (#12976) @noraleonte

#### `@mui/x-date-pickers-pro@7.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@7.4.0`.

### Charts

>  

Reviewer: ```suggestion

```

> ### Charts

Reviewer: ```suggestion
### Charts

```

#### `@mui/x-charts@7.4.0`

- [charts] Add `ChartsGrid` to `themeAugmentation` (#13026) @noraleonte
- [charts] Support rounded corners on `BarChart` (#12834) @JCQuintas

### Tree View

#### `@mui/x-tree-view@7.4.0`

- [TreeView] Fix props propagation and theme entry in `TreeItem2` (#12889) @flaviendelangle

### Docs

- [docs] Add accessibility page to TreeView docs (#12845) @noraleonte

> - [docs] Fix legal link to EULA free trial (#13013) @oliviertassinari

Reviewer: ```suggestion
- [docs] Fix Charts styling typos (#13061) @oliviertassinari
- [docs] Fix legal link to EULA free trial (#13013) @oliviertassinari
- [docs] Update interface name in pinned columns docs (#13070) @cherniavskii
```

### Core

- [core] Improve release process docs (#12977) @JCQuintas
- [core] Prepare React 19 (#12991) @oliviertassinari
- [docs-infra] Fix Netlify PR preview path (#12993) @oliviertassinari
- [infra] Automation: Add release PR reviewers (#12982) @michelengelen

> 

Reviewer: ```suggestion
```

> - [core] Prepare React 19 (#12991) @oliviertassinari

Reviewer: ```suggestion
- [core] Prepare React 19 (#12991) @oliviertassinari
- [docs-infra] Fix custom API page headings (#13074) @LukasTy
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.4.0

> _May 8, 2024_

Reviewer 3: ```suggestion
_May 9, 2024_
```

We'd like to offer a big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:

> - 🎁 Add optional `id` attribute on shortcut items of the Date and Time Pickers

Reviewer 4: Super nitpick: can we use a different emoji? 🙈 
```suggestion
- ✨ Add optional `id` attribute on shortcut items of the Date and Time Pickers
```

- 🎁 Add support for Jalaali v3 in the Date and Time Pickers
- 🐞 Bugfixes
- 📚 Documentation improvements

### Data Grid

#### `@mui/x-data-grid@7.4.0`

> - [DataGrid] Fix throw error with onBlur at SVG element (#13028) @oukunan

Reviewer 3: ```suggestion
- [DataGrid] Fix error when focus moves from column header to `svg` element (#13028) @oukunan
```

> - [DataGrid] Fix: column group change bug (#12965) @romgrk

Reviewer 3: ```suggestion
- [DataGrid] Fix error on column groups change (#12965) @romgrk
```

#### `@mui/x-data-grid-pro@7.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.4.0`.

#### `@mui/x-data-grid-premium@7.4.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.4.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@7.4.0`

- [pickers] Add `AdapterDateFnsJalaliV3` adapter (#12891) @smmoosavi
- [pickers] Add optional `id` attribute on shortcut items (#12976) @noraleonte

#### `@mui/x-date-pickers-pro@7.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@7.4.0`.

### Charts
 
#### `@mui/x-charts@7.4.0`

- [charts] Add `ChartsGrid` to `themeAugmentation` (#13026) @noraleonte

> - [charts] Support rounded corners on `BarChart` (#12834) @JCQuintas

Reviewer 4: This should be a highlight I think 🤔

### Tree View

#### `@mui/x-tree-view@7.4.0`

- [TreeView] Fix props propagation and theme entry in `TreeItem2` (#12889) @flaviendelangle

### Docs

> - [docs] Add accessibility page to TreeView docs (#12845) @noraleonte

Reviewer 4: Another potential highlight 🤔

- [docs] Fix legal link to EULA free trial (#13013) @oliviertassinari

### Core

- [core] Improve release process docs (#12977) @JCQuintas
- [core] Prepare React 19 (#12991) @oliviertassinari

> - [core] Update license key used for tests (#13052) @LukasTy

Reviewer: I'd happily reduce the visibility of this change. 🥷 
WDYT? 🤔 
```suggestion
```

- [docs-infra] Fix Netlify PR preview path (#12993) @oliviertassinari
- [infra] Automation: Add release PR reviewers (#12982) @michelengelen

> - [infra] Changed action target branch (#13029) @michelengelen
> - [infra] Fix automated reviewers assignment workflow (#12986) @michelengelen
> - [infra] Fix automated reviewers assignment workflow - part 2 (#12989) @michelengelen
> - [infra] Fix automated reviewers assignment workflow - part 3 (#13012) @michelengelen
> - [infra] changed permission for nested workflow (#13024) @michelengelen
> - [infra] changed to usage of reusable workflow (#13017) @michelengelen
> - [infra] changed version to use master branch for now (#13022) @michelengelen

Reviewer: WDYT about reducing the "noise" here? 🤔 
```suggestion
```

Reviewer 5: Yes please.... remove them altogether! It's kind of a failed attempt due to permission issues for the GH token! 🤷🏼
